open-metadata-implementation/access-services/data-engine/data-engine-server/src/main/java/org/odpi/openmetadata/accessservices/dataengine/server/handlers/ProcessHandler.java

```diff
@@ -77,17 +79,16 @@ public ProcessHandler(String serviceName, String serverName, InvalidParameterHan
      */
     public String createProcess(String userId, String qualifiedName, String processName, String description,
                                 String latestChange, List<String> zoneMembership, String displayName, String formula,
-                                String owner, OwnerType ownerType) throws InvalidParameterException,
-                                                                          UserNotAuthorizedException,
-                                                                          PropertyServerException {
+                                String owner, OwnerType ownerType, String externalSourceGUID, String externalSourceName)
+            throws InvalidParameterException, UserNotAuthorizedException, PropertyServerException {
         final String methodName = "createProcess";
 
         InstanceProperties properties = buildProcessInstanceProperties(userId, qualifiedName, processName, description,
                 latestChange, zoneMembership, displayName, formula, owner, ownerType, methodName);
 
         TypeDef entityTypeDef = repositoryHelper.getTypeDefByName(userId, ProcessPropertiesMapper.PROCESS_TYPE_NAME);
-        return repositoryHandler.createEntity(userId, entityTypeDef.getGUID(), entityTypeDef.getName(), properties,
-                InstanceStatus.DRAFT, methodName);
+        return repositoryHandler.createExternalEntity(userId, entityTypeDef.getGUID(), entityTypeDef.getName(), externalSourceGUID,
```

Comment:
I notice in most (all?) places we're sending both the external source GUID and the external source name -- do we need both?  If the OMAS is handling this for us in all cases, then no problem; but otherwise it might be easier to ask external engines to provide one or the other (rather than both) (?)

Reply:
I was also a bit unsure about whether I should include both of them in all methods. 
In the original methods from OMRSMetadataCollection it requires external GUID and Name (unique name) when metadata being created from external sources. From OMAS I think it would be good to keep the consistency from that. That's the reason that I did in this way. 

The implementation actually is being done slightly different from DE OMAS spring module and client module: 
If you want to call the DE OMAS REST APIs directly you need to include the external GUID and NAME in every payload. 
If you call it from client side, you need to the method of createExternalDataEngine to create the external engine GUID and Name, which will be save as DE Engine OMAS client side properties. So the following REST calls has been set to use the external source GUID and name from that client instance. In this case the payloads from external DEs can exclude the information with source GUID and name.

Not sure it is the ideal approach? I just did in this way when I was coding it yesterday.

Reply:
Fair enough — let’s discuss briefly in today’s call and see what everyone thinks

Reply:
After today's call we concluded that we should only send the external source Name as a parameter for the methods, and retrieve the GUID of the source from the reporitory based on that name. This should be done internally in DE OMAS server.
The client should also follow the same approach as the spring module, and not save the GUID and Name as properties

Reply:
One minor revision: I think the idea is that the client should expect to receive the name and use this to lookup the GUID, and can "cache" this internally so that it does not need to lookup the GUID on every single request -- but we should expect that the client at some point will be restarted (losing the cache) and that it therefore needs to lookup again based on the name (not create a new external source with the same name / different GUID each time it restarts, but also not expect to always have the GUID and name locally from the very start).

Reply:
My understanding was  from the discussion that the REST API took the name and the server did the look up.  This way the client is effectively stateless.  This is good for an API where the external Id is an occasional use.  However for the OMASs that typically support an engine where the external Id is the same on every request, then I can see the value of caching in the client.

Reply:
Thanks a lot. I will write a conclusion in the issue and redo the logic of this part.